### PR TITLE
Add models with no-ic [training and LOIO]

### PR DESCRIPTION
**Pull request by gwaybio**
@roshankern can you review my contributions (the last two commits), and I will review your contributions (first two commits).

My plan is to get this merged in, address comments in #47, merge #47, and then start a new visualization with a clean repo.

I looked at the results briefly, and they don't seem to be changed much (i.e. IC doesn't seem to have a big impact in this case).

### CODE REVIEW

File: 0.download_data/README.md
Comment (gwaybio) [inline 0.download_data/README.md:17]: why is it necessary to specify for the IC dataset? Isn't it the same for no-ic?

File: 0.download_data/scripts/nbconverted/download_data.py
Comment (gwaybio) [inline 0.download_data/scripts/nbconverted/download_data.py]: is there a place for this data on this repo? It would be great to avoid specific paths that will break on other machines!

File: 1.split_data/scripts/nbconverted/split_data.py
Comment (gwaybio) [inline 1.split_data/scripts/nbconverted/split_data.py]: constants should be defined sooner in the notebook (and also not defined within a loop!)

File: 2.train_model/README.md
Comment (gwaybio) [inline 2.train_model/README.md:36]: :tada:

File: 2.train_model/scripts/nbconverted/train_multi_class_models.py
Comment (gwaybio) [inline 2.train_model/scripts/nbconverted/train_multi_class_models.py:92]: quick clarification: is this where the no-ic vs. ic dataset is being specified? I don't see where this variable is specified.

Comment (gwaybio) [reply 2.train_model/scripts/nbconverted/train_multi_class_models.py:92]: oooh, I see it is actually in line 85!

**Review by gwaybio — COMMENTED** [review]
:+1: on my end! LMK when you're able to review my last two commits. Thanks!

File: 0.download_data/README.md
Comment (roshankern) [reply 0.download_data/README.md:17]: I believe they differ (very slightly). Because illumination correction has a very small effect on segmentation, we get different cell locations and in some cases a different number of cells.

I was going to update this when I update the path that we download the `no_ic` data from in https://github.com/WayScience/mitocheck_data/pull/37. See below for more explanation on this

File: 0.download_data/scripts/nbconverted/download_data.py
Comment (roshankern) [reply 0.download_data/scripts/nbconverted/download_data.py]: I wanted to keep it very clear that this path needs to be updated to reflect the path to data on GitHub in `mitocheck_data`. In an effort to not block you, I worked on the repos in the wrong order (`phenotypic_profiling` then `mitocheck_data`). But once https://github.com/WayScience/mitocheck_data/pull/37 is merged we can update the path in this code. Sorry for the confusion!

File: 1.split_data/scripts/nbconverted/split_data.py
Comment (roshankern) [reply 1.split_data/scripts/nbconverted/split_data.py]: Good catch, fixed!

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py
Comment (roshankern) [inline 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py:137]: Just added a small comment to the readme for this module explaining this.

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py
Comment (roshankern) [inline 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]: Is there a reason to not save this file as `compiled_LOIO_probabilites.tsv` and delete the rest of the LOIO results files (`compiled_LOIO_probabilites_withshuffled.tsv` and `compiled_LOIO_probabilites_withshuffled_withnoic.tsv`). This file should include everything (shuffled, no ic) so I think to avoid confusion it would be best to just have this one file in the repo.

It would also help maintain consistency with the other evaluate results. For example, `F1_scores/` has one file (`compiled_F1_scores.tsv`) with all the parameter combos (data_split	shuffled	feature_type	balance_type	dataset_type).

**Review by roshankern — APPROVED** [review]
Just one comment to address, then LGTM!

Your plan for merging/addressing comments sounds good. I would recommend waiting to merge this until after we merge https://github.com/WayScience/mitocheck_data/pull/37 and integrate the new download link so we don't have an unclear download path on `main`.

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]: this is a good idea. I was previously keeping with your convention of adding `_withshuffled` for example. I will update the file name and remove other files, thanks!

**Comment by gwaybio** [comment]
> I would recommend waiting to merge this until after we merge https://github.com/WayScience/mitocheck_data/pull/37 and integrate the new download link so we don't have an unclear download path on main.

Sounds good, thanks @roshankern ! Please give me a 👍 when I can merge. (I will also fix your file name comment ASAP).

**Comment by roshankern** [comment]
> Sounds good, thanks @roshankern ! Please give me a 👍 when I can merge. (I will also fix your file name comment ASAP).

Will do!

**Comment by gwaybio** [comment]
#37 looks great @roshankern, I just reviewed. I think there is one small change to do.

I also addressed the file name fix you suggested above. Once you merge in #37 feel free to merge this one.

File: 0.download_data/scripts/nbconverted/download_data.py
Comment (roshankern) [reply 0.download_data/scripts/nbconverted/download_data.py]: Fixed!

**Pull request merged**
